curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -67,44 +67,64 @@ void FsCacheManager::ReleaseFileCacheManager(uint64_t inodeId) {
     return;
 }
 
-std::pair<std::list<DataCachePtr>::iterator, WeakDataCachePtr>
-FsCacheManager::Set(DataCachePtr dataCache) {
-    WriteLockGuard writeLockGuard(rwLockLru_);
+std::list<DataCachePtr>::iterator FsCacheManager::Set(DataCachePtr dataCache) {
+    std::lock_guard<std::mutex> lk(lruMtx_);
 
     VLOG(3) << "lru current byte:" << lruByte_
             << ",lru max byte:" << readCacheMaxByte_;
 
     // trim cache without consider dataCache's size, because its size is
     // expected to be very smaller than `readCacheMaxByte_`
-    while (lruByte_ >= readCacheMaxByte_) {
-        auto& trimDataCache = lruReadDataCacheList_.back();
-        VLOG(3) << "lru release data cache, size: "
-        << trimDataCache->GetLen();
-        lruByte_ -= trimDataCache->GetLen();
-        trimDataCache->Release();
-        lruReadDataCacheList_.pop_back();
+    if (lruByte_ >= readCacheMaxByte_) {
```

Comment:
if is redundant？
{
      uint64_t retiredBytes = 0;
      auto iter = lruReadDataCacheList_.end();
      while(){}
}

Reply:
not really, if do what you say, another if should be added to check whether I have to cut some data chches from lruReadDataCacheList_ or not, because while loop does not pop any data cache.